crates/orchestrator/src/tests/jobs/state_update_job/mod.rs

```diff
@@ -21,16 +21,89 @@ use crate::jobs::{
+use crate::jobs::state_update_job::utils::{fetch_blob_data_for_block, hex_string_to_u8_vec};
+use crate::tests::common::{default_job_item, get_storage_client};
+use crate::tests::config::TestConfigBuilder;
 use httpmock::prelude::*;
 use lazy_static::lazy_static;
+use utils::env_utils::get_env_var_or_panic;
 
 lazy_static! {
     pub static ref CURRENT_PATH: PathBuf = std::env::current_dir().unwrap();
 }
 
 pub const X_0_FILE_NAME: &str = "x_0.txt";
 
+// ================= Exhaustive tests (with minimum mock) =================
+
+#[rstest]
+#[should_panic(expected = "Could not find current attempt number.")]
+#[tokio::test]
+async fn test_process_job_attempt_no_not_present_fails() {
+    TestConfigBuilder::new().build().await;
+
+    let mut job = default_job_item();
+    let config = config().await;
+    let state_update_job = StateUpdateJob {};
+    let _ = state_update_job.process_job(&config, &mut job).await;
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_process_job_works() {
+    // Will be used by storage client which we call while storing the data.
+    dotenvy::from_filename("../.env.test").expect("Failed to load the .env file");
+
+    // Mocking the settlement client.
+    let mut settlement_client = MockSettlementClient::new();
+
+    let block_numbers: Vec<u64> = vec![651053, 651054, 651055];
+    // This must be the last block number and should be returned as an output from the process job.
+    let last_block_number = "651055";
+
+    // Storing `blob_data` and `snos_output` in storage client
+    store_data_in_storage_client_for_s3(block_numbers.clone()).await;
+
+    // Building a temp config that will be used by `fetch_blob_data_for_block` and `fetch_snos_for_block`
+    // functions while fetching the blob data from storage client.
+    TestConfigBuilder::new().build().await;
+
+    // Adding expectations for each block number to be called by settlement client.
+    for i in block_numbers {
+        let blob_data = fetch_blob_data_for_block(i).await.unwrap();
+        settlement_client
+            .expect_update_state_with_blobs()
+            .with(eq(vec![]), eq(blob_data))
+            .times(1)
+            .returning(|_, _| Ok("0xbeef".to_string()));
+    }
+    settlement_client.expect_get_last_settled_block().with().returning(|| Ok(651052));
+
+    // Building new config with mocked settlement client
+    TestConfigBuilder::new().mock_settlement_client(Box::new(settlement_client)).build().await;
+
+    // setting last failed block number as 651053.
+    // setting blocks yet to process as 651054 and 651055.
+    // here total blocks to process will be 3.
+    let mut metadata: HashMap<String, String> = HashMap::new();
+    metadata.insert(JOB_PROCESS_ATTEMPT_METADATA_KEY.to_string(), "0".to_string());
+    metadata.insert(JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO.to_string(), "651053".to_string());
+    metadata.insert(JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY.to_string(), "651053,651054,651055".to_string());
```

Comment:
can we do cases for this

1. no failed block no present
2. failed block present (in the middle so that we can check that processed ones weren't processed again)

Reply:
Added the test cases.